source/common/upstream/cluster_factory_impl.cc

```diff
@@ -108,7 +108,7 @@ ClusterFactoryImplBase::create(const envoy::config::cluster::v3::Cluster& cluste
       new_cluster_pair.first->setHealthChecker(HealthCheckerFactory::create(
           cluster.health_checks()[0], *new_cluster_pair.first, server_context.runtime(),
           server_context.mainThreadDispatcher(), server_context.accessLogManager(),
-          context.messageValidationVisitor(), server_context.api()));
+          context.messageValidationVisitor(), server_context.api(), server_context));
```

Comment:
It looks like all parameters that are initialized from server_context properties can be removed and just use `server_context` member variable later on. Would it be too much work to clean these up?

Reply:
Done.